gremlin-go/driver/client.go

```diff
@@ -71,11 +73,20 @@ func NewClient(url string, configurations ...func(settings *ClientSettings)) (*C
 		transporterType: settings.TransporterType,
 		connection:      conn,
 	}
+	// TODO: PoolSize must be 1 on Session mode
 	return client, nil
 }
 
 // Close closes the client via connection.
 func (client *Client) Close() error {
+	// If it is a Session, call closeSession
+	if client.session != "" {
+		err := client.closeSession()
+		if err != nil {
+			return err
+		}
+	}
+	client.logHandler.logf(Info, closeClient, client.url)
 	return client.connection.close()
 }
```

Comment:
You probably need something like this @L0Lmaker 

```suggestion
func (client *Client) Close() error {
	// If it is a Session, call closeSession
	var sessionError error
	if client.session != "" {
		err = client.closeSession()
	}
	client.logHandler.logf(Info, closeClient, client.url)
	closeError := client.connection.close()
	// TODO: Combine sessionError and closeError  and return if both are not nil.
}
```